src/main/kotlin/org/jitsi/videobridge/transport/udp/UdpTransport.kt

```diff
@@ -0,0 +1,213 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.transport.udp
+
+import org.ice4j.socket.SocketClosedException
+import org.jitsi.nlj.util.OrderedJsonObject
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.createChildLogger
+import org.jitsi.utils.stats.RateStatistics
+import java.io.IOException
+import java.net.DatagramPacket
+import java.net.DatagramSocket
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.net.SocketAddress
+import java.net.SocketException
+import java.net.UnknownHostException
+import java.time.Clock
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * A transport which allows sending and receiving data over UDP.  The transport
+ * will bind to [bindPort] on [bindAddress] and will listen for incoming
+ * packets there via [startReadingData] until the transport is stopped. Sending
+ * can be done via [send], and a remote address (or a set of remote addresses)
+ * must be provided.
+ */
+class UdpTransport @JvmOverloads @Throws(SocketException::class, UnknownHostException::class) constructor(
+    private val bindAddress: String,
+    private val bindPort: Int,
+    parentLogger: Logger,
+    soRcvBuf: Int? = null,
+    soSndBuf: Int? = null,
+    private val clock: Clock = Clock.systemUTC()
+) {
+    private val logger = createChildLogger(parentLogger, mapOf(
+        "address" to bindAddress,
+        "port" to bindPort.toString()
+    ))
+
+    private val running = AtomicBoolean(true)
+
+    private val socket: DatagramSocket = DatagramSocket(
+        InetSocketAddress(InetAddress.getByName(bindAddress), bindPort)
+    ).apply {
+        soRcvBuf?.let { receiveBufferSize = it }
+        soSndBuf?.let { sendBufferSize = it }
+    }.also { socket ->
+        logger.info("Initialized with bind address $bindAddress and bind port $bindPort. " +
+                "Receive buffer size ${socket.receiveBufferSize}${soRcvBuf?.let { " (asked for $it)"} ?: ""}. " +
+                "Send buffer size ${socket.sendBufferSize}${soSndBuf?.let { " (asked for $it)"} ?: ""}.")
+    }
+
+    val stats = Stats()
+
+    var incomingDataHandler: IncomingDataHandler? = null
+
+    /**
+     * Read data for as long as this transport is still running.  Received data
+     * is passed to the set [IncomingDataHandler].
+     */
+    fun startReadingData() {
+        val buf = ByteArray(1500)
+        val packet = DatagramPacket(buf, 0, 1500)
+        while (running.get()) {
+            // TODO: needed?
+            packet.setData(buf, 0, 1500)
+            try {
+                socket.receive(packet)
+                clock.instant().also { now ->
```

Comment:
Is ther a reason this block is inside `try`?

Personal preference: I find this easier to read
```
val now = clock.instant()
stats.packetReceived(...)
...
```

Reply:
Fixed